spark/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala

```diff
@@ -133,6 +134,25 @@ class DeltaReorgSuite extends QueryTest
     }
   }
 
+  test("REORG TABLE UPGRADE UNIFORM from v1 to v2") {
```

Comment:
In addition, we need to move this test to suites like [UniFormE2ESuite](https://github.com/delta-io/delta/blob/master/iceberg/src/test/scala/org/apache/spark/sql/delta/uniform/UniFormE2ESuite.scala) so that we could use `HMS` rather than the default delta lake catalog, which does not natively support storing the iceberg metadata.

I could help with the test case as I'm currently adding the test suites related to UniForm and `REORG UPGRADE UNIFORM` is also included.

cc @KaiqiJinWow.